src/Core/Services/AuthService.cs

```diff
@@ -346,6 +346,7 @@ private async Task<SymmetricCryptoKey> MakePreloginKeyAsync(string masterPasswor
                 TwoFactorProvidersData = response.TwoFactorResponse.TwoFactorProviders2;
                 result.TwoFactorProviders = response.TwoFactorResponse.TwoFactorProviders2;
                 CaptchaToken = response.TwoFactorResponse.CaptchaToken;
+                await _tokenService.ClearTwoFactorTokenAsync(email);
```

Comment:
I _think_ this may be unneccesary once the other change is made in `StateService`.  Once that's done can you remove this and re-test?